api/server/sdk/credentials.go

```diff
@@ -229,7 +237,15 @@ func (s *CredentialServer) Delete(
 		return nil, status.Error(codes.InvalidArgument, "Must provide credentials uuid")
 	}
 
-	err := s.driver(ctx).CredsDelete(req.GetCredentialId())
+	// Check ownership
+	_, err := s.Inspect(ctx, &api.SdkCredentialInspectRequest{
```

Comment:
Seems like Inspect only checks for Read access. However Delete must be an admin access right?